toolkit/ar/src/main/java/com/arcgismaps/toolkit/ar/internal/ArSurfaceView.kt

```diff
@@ -0,0 +1,109 @@
+/*
+ *
+ *  Copyright 2024 Esri
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package com.arcgismaps.toolkit.ar.internal
+
+import android.content.Context
+import android.opengl.GLSurfaceView
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.DisposableEffect
+import androidx.compose.runtime.remember
+import androidx.compose.ui.ExperimentalComposeUiApi
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.input.pointer.pointerInteropFilter
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.LocalLifecycleOwner
+import androidx.compose.ui.viewinterop.AndroidView
+import androidx.lifecycle.DefaultLifecycleObserver
+import androidx.lifecycle.LifecycleOwner
+import com.arcgismaps.toolkit.ar.ArSessionWrapper
+import com.arcgismaps.toolkit.ar.internal.render.CameraFeedRenderer
+import com.arcgismaps.toolkit.ar.internal.render.SurfaceDrawHandler
+import com.google.ar.core.Frame
+import com.google.ar.core.HitResult
+import com.google.ar.core.Session
+
+/**
+ * Renders the AR camera feed using a [GLSurfaceView].
+ *
+ * @param arSessionWrapper an [ArSessionWrapper] that provides an ARCore [Session].
+ * @param onFrame a callback that is invoked every frame.
+ * @param onTap a callback that is invoked when the user taps the screen and a hit is detected.
+ * @since 100.6.0
+ */
+@OptIn(ExperimentalComposeUiApi::class)
+@Composable
+internal fun ARSurfaceView(
```

Comment:
A more descriptive name might be?
```suggestion
internal fun ArCameraFeed(
```